test/unit/specs/main.spec.js

```diff
@@ -173,26 +173,32 @@ describe("Startup Process", () => {
       expect(main.processes.lcdProcess).toBeDefined()
     })
 
-    it("should use a provided node-ip to connect to", async function() {
+    it("should use a provided node-ip to connect to", async function(done) {
       main.shutdown()
       prepareMain()
 
+      jest.doMock(
+        "app/src/main/addressbook.js",
+        () =>
+          class MockAddressbook {
+            constructor(path, peers) {
+              try {
+                expect(peers).toEqual(["123.456.789.123"])
+                done()
+              } catch (err) {
+                done.fail(err)
+              } finally {
+                delete process.env.COSMOS_NODE
+              }
+            }
+          }
+      )
+
       Object.assign(process.env, {
         COSMOS_NODE: "123.456.789.123"
       })
       // run main
-      main = await require(appRoot + "src/main/index.js")
-
-      expect(
-        childProcess.spawn.mock.calls.find(
-          ([path, args]) =>
-            path.includes("gaiacli") &&
-            args.includes("rest-server") &&
-            args.includes("123.456.789.123:46657")
-        )
-      ).toBeDefined()
```

Comment:
why would these not pass anymore?

Reply:
I put this logic now into the addressbook